src/net/ossl.cc

```diff
@@ -1054,6 +1054,19 @@ class session : public enable_shared_from_this<session>, public session_impl {
                     // return an empty buffer (the get() function will initiate a handshake)
                     return make_ready_future<buf_type>();
                 case SSL_ERROR_SYSCALL:
+                    if (errno == 0 && ERR_peek_error() == 0) {
+                        // SSL_get_error
+                        // (https://www.openssl.org/docs/man3.0/man3/SSL_get_error.html)
+                        // states that on OpenSSL versions prior to 3.0, an
+                        // SSL_ERROR_SYSCALL with nothing on the stack and errno
+                        // == 0 indicates EOF but future versions should report
+                        // SSL_ERROR_SSL with SSL_R_UNEXPECTED_EOF_WHILE_READING
+                        // on the stack.  However we are seeing situations where
+                        // SSL_ERROR_SYSCALL is returned and errno == 0 and the
```

Comment:
suggestion(async): Might want to make a note of ossl versions that are known to exhibit this behavior.

Reply:
oh great point